src/Sylius/Bundle/CoreBundle/Resources/config/app/workflow/sylius_catalog_promotion.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+use Sylius\Component\Core\Model\CatalogPromotion;
+use Symfony\Config\FrameworkConfig;
+
+return static function (FrameworkConfig $framework): void {
+    $catalogPromotion = $framework->workflows()->workflows('sylius_catalog_promotion');
+    $catalogPromotion
+        ->type('state_machine')
+        ->supports([CatalogPromotion::class])
```

Comment:
Parameter class does not work for now. Let it as is for now, and a task to resolve these resource classes.

Reply:
What about the interface? Is there anything else injected into this callable? Do you think that there may be a problem with the order of loading of bundles?

Reply:
I don't know why it does not work, we have to investigate. I'm sure I've tried to change the order of bundles.